python/semantic_kernel/connectors/memory/azure_cosmosdb_no_sql/azure_cosmos_db_no_sql_collection.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+import sys
+from collections.abc import Sequence
+from typing import Any, TypeVar
+
+if sys.version_info >= (3, 12):
+    from typing import override  # pragma: no cover
+else:
+    from typing_extensions import override  # pragma: no cover
+
+from azure.cosmos.exceptions import CosmosResourceNotFoundError
+from azure.cosmos.partition_key import PartitionKey
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.memory.azure_cosmosdb_no_sql.azure_cosmos_db_no_sql_base import AzureCosmosDBNoSQLBase
+from semantic_kernel.connectors.memory.azure_cosmosdb_no_sql.azure_cosmos_db_no_sql_composite_key import (
+    AzureCosmosDBNoSQLCompositeKey,
+)
+from semantic_kernel.connectors.memory.azure_cosmosdb_no_sql.azure_cosmos_db_no_sql_settings import (
+    AzureCosmosDBNoSQLSettings,
+)
+from semantic_kernel.connectors.memory.azure_cosmosdb_no_sql.utils import (
+    COSMOS_ITEM_ID_PROPERTY_NAME,
+    build_query_parameters,
+    create_default_indexing_policy,
+    create_default_vector_embedding_policy,
+    get_key,
+    get_partition_key,
+)
+from semantic_kernel.data.vector_store_model_definition import VectorStoreRecordDefinition
+from semantic_kernel.data.vector_store_record_collection import VectorStoreRecordCollection
+from semantic_kernel.exceptions.memory_connector_exceptions import (
+    MemoryConnectorException,
+    MemoryConnectorInitializationError,
+    MemoryConnectorResourceNotFound,
+    VectorStoreModelDeserializationException,
+)
+from semantic_kernel.kernel_types import OneOrMany
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+TModel = TypeVar("TModel")
+TKey = TypeVar("TKey", str, AzureCosmosDBNoSQLCompositeKey)
+
+
+@experimental_class
+class AzureCosmosDBNoSQLCollection(AzureCosmosDBNoSQLBase, VectorStoreRecordCollection[TKey, TModel]):
```

Comment:
needs explicit Generic[Tkey, TModel] AFAIK

Reply:
Will that not inherit from `VectorStoreRecordCollection[TKey, TModel]`?